### PR TITLE
[API Management] `az apim api export`: Add command to export an API Management API

### PR DESCRIPTION
**Pull request by arpishahmsft**
**Related command**

az apim api export -g Api-Default-Central-US-EUAP --service-name arpishahtestapim --api-id echo-api --format OpenApiJson --file-path C:\Users\arpishah\examples\cli-examples\exported-results\exported-apim-results

**Description**

This PR is to add export apim cli command. The import command was already present but the export command was not present. This change exports an API present in the APIM instance to a file based on the format specified or to a link. This is controlled by the format parameter

**Testing Guide**

az apim api export -g Api-Default-Central-US-EUAP --service-name arpishahtestapim --api-id echo-api --format OpenApiJson --file-path C:\Users\arpishah\examples\cli-examples\exported-results\exported-apim-results 
Here the api echo-api present in arpishahtestapim apim instance is exported to the file C:\Users\arpishah\examples\cli-examples\exported-results\exported-apim-results --subscription-id a200340d-6b82-494d-9dbf-687ba6e33f9e in the format openapi json

**History Notes**

 `az apim api export: Add export api from apim capability

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/apim/custom.py [hunk 1]
```diff
@@ -513,6 +515,108 @@ def apim_api_import(
...
+        logger.warning(f"Error writing to file.: {e}")
+    except Exception as e:
+        logger.warning(f"An error occurred: {e}")
+    
+    # Write the response to a file
+    return logger.warning(f"APIMExport results written to file: {full_path}")
+
+def api_export_result_to_dict(api_export_result):
+    # This function should return a dictionary representation of the ApiExportResult object
+    # The implementation of this function depends on the structure of the ApiExportResult object
```
Comment (arpishahmsft) [inline src/azure-cli/azure/cli/command_modules/apim/custom.py:613]: Remove this comment

**Comment by yanzhudd** [comment]
please fix the CI issues

**Comment by yanzhudd** [comment]
Please not that CLI only review the code style and specification, but not responsible for business logic.
Thus, it is better to involve service team to review PRs as well.

**Comment by yingru97** [comment]
@arpishahmsft The PR looks good to me

**Comment by arpishahmsft** [comment]
[yanzhudd](https://github.com/yanzhudd) The PR has been reviewed by APIM Team member - Alan [yingru97](https://github.com/yingru97)

Alan, however, is not able to approve PR, he does not get an option to join as a reviewer. 
Could you please help hoim add as a reviewer or approve this PR as its also reviewed by Alan. Thanks!

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**